docs/specifications/tests/MethodsV2.md

```diff
@@ -0,0 +1,1026 @@
+
+This is an [Internal Method][Internal Methods] that can be referred to by other
+Methods in this document, but not by Test Case specifications.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+
+### Prerequisite
+
+None.
+
+### Test procedure
+
+1. Using Method [Get-Del-NS-IPs], obtain the IP addresses to the name
+   servers ("Name Server IPs").
+
+2. Using Method [Get-Zone-NS-Names], obtain the names of the name servers
+   from the *Child Zone* ("Child Zone Name Server Names").
+
+3. If the *Name Server IPs* set or the *Child Zone Name Server Names* set is
+   empty or undefined, then output an undefined set and exit these test
+   procedures.
+
+3. If no name in *Child Zone Name Server Names* is an [in-bailiwick]
+   name server name:
+   1. Output an empty set.
+   2. Exit these test procedures.
+
+4. Create an empty set the [in-bailiwick] name server names from the
+   *Child Zone Name Server Names* set, where each name is linked to an empty set
+   of IP addresses ("Name Servers").
+
+5. For name in *Name Servers* do:
+   1. Create the following two [DNS queries][DNS Query]:
+      1. Query type A and the [in-bailiwick] name as the query name ("A Query").
+      2. Query type AAAA and the [in-bailiwick] name as the query name
+         ("AAAA Query").
+   2. [Send] *A Query* and *AAAA Query* to all servers in *Name Server IPs*
+      and process the [DNS Responses][DNS Response] from each of them.
+   3. If a delegation (referral) to a sub-zone of Child Zone is returned,
+      follow that delegation, possibly in several steps, by repeating
+      *A Query* and *AAAA Query*.
+   4. If a CNAME is returned, follow that, possibly in several
+      steps, to resolve the name to IP addresses, if possible.
+   5. Ignore non-referral responses unless AA flag is set (cached data
+      is not accepted) and ignore response with any other [RCODE Name] than
+      NoError.
+   6. Add found IP addresses for the name server names in *Name Servers*.
+
+6. Output the possibly empty *Name Servers* set.
+
+### Outputs
+
+* A set of name server names pointing at possibly empty sets of IP addresses:
+  * Non-empty set: The normal case.
+  * Empty set: There are no [in-bailiwick] names or those are not defined in
+    *Child Zone*, also a normal case.
+  * Undefined set: [Get-Del-NS-IPs] returned an empty or undefined set.
+
+### Special procedural requirements
+
+None.
+
+### Dependencies
+
+This Method depends on [Get-Zone-NS-Names] and [Get-Del-NS-IPs].
+
+[To top]
+
+
+## Method: Get out-of-bailiwick ip addresses (Internal)
+
+### Method identifier
+**Get-OOB-IPs**
+
+### Objective
+
+Obtain the IP addresses of the out-of-bailiwick name servers for the
+given zone (child zone) and a given set of name server names. Extract
+addresses even if the resolution goes through CNAME. It is,
+however, not permitted for a NS record to point at a name that has a
+CNAME, but that test is covered by Test Case [DELEGATION05].
```

Comment:
I suggest this paragraph is split such that the main purpose is summarized in the first one and the edge case is described in a second one.

Also the wording in the description of the edge case is in terms of what is allowed by the RFCs. I suggest it is reworded in terms of how this method deals with the edge case.

Reply:
Reworded.